lib/dbservice_web/controllers/group_user_controller.ex

```diff
@@ -105,6 +105,35 @@ defmodule DbserviceWeb.GroupUserController do
     end
   end
 
+  def update_by_type(conn, params) do
+    user_id = params["user_id"]
+    type = params["type"]
+    group = Groups.get_group_by_group_id_and_type(params["group_id"], type)
+    new_group_id = group.child_id
+
+    # Fetch the GroupUser with the specified user_id and where the group type is the provided type
+    group_user =
+      GroupUsers.get_group_user_by_user_id_and_type(user_id, type)
+      |> List.first()
+
+    # Fetch the EnrollmentRecord with the specified user_id and where the group_type matches the provided type
+    enrollment_record =
+      from(er in EnrollmentRecord,
+        where: er.user_id == ^user_id and er.group_type == ^type
+      )
+      |> Repo.one()
```

Comment:
Why only one? If it's batch, student can belong to multiple batches no?

Reply:
Hmm okay. Just add a comment to the method that this method assumes that only one ER will be updated so that we don’t anticipate need for multiple updates in same call — just so we don’t lose track of this.

Also add nice doc waala comment to the method :)

Reply:
Okay!